tests/virt_autotest/reboot_and_wait_up.pm

```diff
@@ -60,14 +60,16 @@ sub reboot_and_wait_up {
             ipmi_backend_utils::ipmitool 'chassis power reset';
         }
         else {
-            #login
-            #The timeout can't be too small since autoyast installation
-            assert_screen "text-login", 600;
-            enter_cmd "root";
-            assert_screen "password-prompt";
-            type_password;
-            send_key('ret');
-            assert_screen "text-logged-in-root";
+            #login is required when sol console is used for the first time
+            unless (check_screen('text-logged-in-root')) {
+                #The timeout can't be too small since autoyast installation
+                assert_screen "text-login", 600;
+                enter_cmd "root";
+                assert_screen "password-prompt";
+                type_password;
+                send_key('ret');
+                assert_screen "text-logged-in-root";
+            }
```

Comment:
What about 
```
elsif (! check_screen('text-logged-in-root')) 
xxxxx
else 
yyyyy
```

Reply:
No actition is needed for 'else', so there is no 'else' here.  Could you give an example if I did not get your meaning exactly?

Reply:
I do not know either. After thinking twice, I think your logic here is enough.